src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java

```diff
@@ -135,14 +160,16 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnul
         List<String> logFile = IOUtils.readLines(run.getLogReader());
         String analysisUrl = analysisService.getAnalysisUrlFromLogFile(logFile);
 
-        if (analysisUrl == null) {
-            analysisUrl = analysisService.getLatestAnalysisUrlForAProject(baseHubUri, expandedProjectName);
+        String aid = this.getAid(run, workspace, expandedProjectName);
+        if(aid == null){
+            // Can't get the analysis ID, so let's mark the build as aborted
+            run.setResult(hudson.model.Result.ABORTED);
         }
-        // here we also need to change something
+
         Analysis analysisWarnings = analysisService.getAnalysisFromUrlWarningsByFilter(analysisUrl);
```

Comment:
Shouldn't `analysisUrl` be generated from `aid` and not from the log?  Shouldn't we update `analysisUrl` here based on `aid`?